pkg/devfile/adapters/kubernetes/component/adapter.go

```diff
@@ -411,28 +411,44 @@ func (a *Adapter) createOrUpdateComponent(componentExists bool, ei envinfo.EnvSp
 	labels["component"] = componentName
 	labels[componentlabels.ComponentTypeLabel] = componentType
 
-	containers, err := generator.GetContainers(a.Devfile, parsercommon.DevfileOptions{})
+	allContainers, err := generator.GetContainers(a.Devfile, parsercommon.DevfileOptions{})
 	if err != nil {
 		return err
 	}
 
+	var initContainers []corev1.Container
```

Comment:
the fact that we have to do this speaks of a need for improvement on the library side. We should be able to fetch init containers and the remaining containers seperately. 
Nice one though

Reply:
Right, the devfile team has made a change on the library for this recently: https://github.com/devfile/library/pull/108